main.js

```diff
@@ -1884,33 +1898,41 @@ else {
     })
 
     app.whenReady().then(async () => {
-        checkWindowPosition(
+        var adjustedWindow = await checkWindowPosition(
             settingsProvider.get('window-position'),
             settingsProvider.get('window-size')
         )
-            .then((visiblePosition) => {
-                console.log(visiblePosition)
-                settingsProvider.set('window-position', visiblePosition)
-            })
-            .catch(() => {})
+        if (adjustedWindow) {
+            console.log(adjustedWindow)
+            settingsProvider.set('window-position', adjustedWindow.position)
+            settingsProvider.set('window-size', adjustedWindow.size)
+        }
 
         checkWindowPosition(settingsProvider.get('lyrics-position'), {
             width: 700,
             height: 800,
         })
-            .then((visiblePosition) => {
-                console.log(visiblePosition)
-                settingsProvider.set('lyrics-position', visiblePosition)
+            .then((adjustedWindow) => {
+                console.log(adjustedWindow)
+                settingsProvider.set('lyrics-position', adjustedWindow.position)
+                settingsProvider.set('lyrics-size', adjustedWindow.size)
```

Comment:
Why are you saving Lyric Size, if you then aren't using it on Lines 1912-1913

Reply:
It was already like that and didn't wanna change it in case whomever did it intentionally wanted to try and run it at the size in those lines.

I merely updated the code to receive a

``` 
adjustedWindow {
    position
    size
}
```

object.

Same for the `checkWindowPosition` right below this one